lms/djangoapps/course_home_api/outline/v1/serializers.py

```diff
@@ -0,0 +1,29 @@
+"""
+Outline Tab Serializers.
+"""
+
+
+from rest_framework import serializers
+
+
+class CourseToolSerializer(serializers.Serializer):
+    """
+    Serializer for Course Tool Objects
+    """
+    analytics_id = serializers.CharField()
+    icon_classes = serializers.CharField()
```

Comment:
I know we talked about this before, but looking at this now... I don't know if I love encoding the known-deprecated FA webfont strings as part of this new API.

I might feel better if (since the MFE is doing a `switch` anyway and only supports certain icons) if we made that part of the contract and abstracted the existence of FA away at all. So the API knows nothing about FA and a theoretical MFE might reasonable just throw in hand drawn icons or whatever.

Something like:
```
def get_icon(self, tool):
    if tool.icon_classes == 'fa fa-notebook-o':
        return 'notebook'
    elif ...
```

And then the API is just returning a well-defined enum that the MFE does its `switch` against.

Is that too engineered? Thoughts, @ciduarte & @Dillon-Dumesnil?

Reply:
I think I'm good with that. We should clearly document what the supported values are for the enum (although having the code might be enough), but otherwise, yeah, I think that sounds reasonable.

Reply:
So it's documented, we'll be removing the icon_classes field from the API. In the MFE, we'll use the analytics_id to determine the icon displayed. https://github.com/edx/frontend-app-learning/pull/73#discussion_r433305435